Formula/gap.rb

```diff
@@ -0,0 +1,137 @@
+  #   with Homebrew), and a number of GAP packages under `pkg/`, some of
+  #   which need to be built.
+
+  option "with-packages",
+         "Try to build included packages using BuildPackages script"
+
+  deprecated_option "with-InstPackages" => "with-packages"
+
+  depends_on "gmp"
+  # NOTE:  A version of [GMP](https://gmplib.org/) is included in GAP archive
+  #   under `extern/`, it is possible to use it instead of the brewed `gmp`.
+  #   See https://www.gap-system.org/Download/INSTALL for details.
+
+  depends_on "readline" => :recommended
+
+  def install
+    # Remove some useless files
+    rm Dir["bin/*.bat", "bin/*.ico", "bin/*.bmp", "bin/cygwin.ver"]
+
+    # Remove GMP archives (`gmp` formula is declared as a dependency)
+    rm Dir["extern/gmp-*.tar.gz"]
+
+    # XXX:  Currently there is no `install` target in `Makefile`.
+    #   According to the manual installation instructions in
+    #
+    #     https://www.gap-system.org/Download/INSTALL ,
+    #
+    #   the compiled "bundle" is intended to be used "as is," and there is
+    #   no instructions for how to remove the source and other unnecessary
+    #   files after compilation.  Moreover, the content of the
+    #   subdirectories with special names, such as `bin` and `lib`, is not
+    #   suitable for merging with the content of the corresponding
+    #   subdirectories of `/usr/local`.  The easiest temporary solution seems
+    #   to be to drop the compiled bundle into `<prefix>/libexec` and to
+    #   create a symlink `<prefix>/bin/gap` to the startup script.
+    #   This use of `libexec` seems to contradict Linux Filesystem Hierarchy
+    #   Standard, but is recommended in Homebrew's "Formula Cookbook."
+
+    args = %W[--prefix=#{libexec} --with-gmp=system]
+
+    args << "--#{build.with?("readline") ? "with" : "without"}-readline"
+    # NOTE: `--with-readline` is the default, it is included for clarity
+
+    system "./configure", *args
+
+    # Fix a bug caused by the buggy `configure` which does not respect
+    # `--prefix` when generating the startup script: the variable `GAP_DIR`
+    # is being set to the (temporary) build directory, but should be set to
+    # the value of `--prefix` option.
+    ["bin/gap-default32.sh", "bin/gap-default64.sh"].each do |startup_script|
+      next unless File.exist?(startup_script)
+      inreplace startup_script, /^GAP_DIR="[^"]*"$/,
+                                        "GAP_DIR=\"#{libexec}\""
+    end
+
+    system "make"
+
+    libexec.install Dir["*"]
+
+    # Create a symlink `bin/gap` from the symlink `gap.sh`
+    cd libexec/"bin" do
+      # NOTE: the symbolic link `gap.sh` is (or may be) relaive
+      bin.install_symlink File.expand_path(`readlink -n gap.sh`) => "gap"
+    end
+
+    if build.with? "packages"
+      ohai "Trying to automatically build included packages"
+      cd libexec/"pkg" do
+        # NOTE:  running this script is known to produce a number of error
```

Comment:
We need to understand, and report these errors.

Reply:
All these errors are well-known to the GAP team (and many will be fixed in the upcoming GAP 4.9. They result from 3rd party packages bundled with GAP sometimes having unclean, unportable or outdated build systems of their own.

You can of course report this back to us, but we'll just close the issue report if it comes from a GAP 4.8 user. If you are concerned about this errors and warnings, I recommend either not building any packages, or only building an explicit subset of packages which is known to work; or to remove packages which are known to produce errors.